### ISSUE TITLE
OOB becomes nan for larger numbers of num_data_points_per_tree

### ISSUE THREAD

**Issue by AndreBiedenkapp** [automl/random_forest_run#36]
Here the results when performing forward-selection with PIMP:

`num_data_points_per_tree=500`
`X.shape (3741, 55)`
> INFO:Forward Selection
Evaluating sp-clause-activity-inc
 OOB: 0.306053
RMSE: 0.298566
>
>Evaluating sp-clause-decay
 OOB: 0.303003
RMSE: 0.296279
>
>Evaluating sp-clause-del-heur
 OOB: 0.337701
RMSE: 0.335855

`num_data_points_per_tree=1500`
`X.shape (3741, 55)`
>INFO:Forward Selection
Evaluating sp-clause-activity-inc
 OOB: 0.311971
RMSE: 0.297653
>
>Evaluating sp-clause-decay
 OOB: 0.300374
RMSE: 0.289780
>
>Evaluating sp-clause-del-heur
 OOB: 0.337702
RMSE: 0.336227

`num_data_points_per_tree=2000`
`X.shape (3741, 55)`
>INFO:Forward Selection
Evaluating sp-clause-activity-inc
 OOB: 0.318140
RMSE: 0.302362
>
>Evaluating sp-clause-decay
 OOB: nan
RMSE: 0.298979
>
>Evaluating sp-clause-del-heur
 OOB: 0.337619
RMSE: 0.336822

`num_data_points_per_tree=3741`
`X.shape (3741, 55)`
>INFO:Forward Selection
Evaluating sp-clause-activity-inc
 OOB: nan
RMSE: 0.277847
>
>Evaluating sp-clause-decay
 OOB: nan
RMSE: 0.281619
>
>Evaluating sp-clause-del-heur
 OOB: nan
RMSE: 0.335883

If I limit the number of datapoints in X:
`num_data_points_per_tree=250`
`X.shape (500, 55)`
>INFO:Forward Selection
Evaluating sp-clause-activity-inc
 OOB: 0.203998
RMSE: 0.338962
>
>Evaluating sp-clause-decay
 OOB: 0.200869
RMSE: 0.330463
>
>Evaluating sp-clause-del-heur
 OOB: 0.208321
RMSE: 0.350660

`num_data_points_per_tree=333`
`X.shape (500, 55)`
>INFO:Forward Selection
Evaluating sp-clause-activity-inc
 OOB: nan
RMSE: 0.348709
>
>Evaluating sp-clause-decay
 OOB: 0.233963
RMSE: 0.343261
>
>Evaluating sp-clause-del-heur
 OOB: 0.234469
RMSE: 0.357129

`num_data_points_per_tree=500`
`X.shape (500, 55)`
>INFO:Forward Selection
Evaluating sp-clause-activity-inc
 OOB: nan
RMSE: 0.331204
>
>Evaluating sp-clause-decay
 OOB: nan
RMSE: 0.322969
>
>Evaluating sp-clause-del-heur
 OOB: nan
RMSE: 0.348493

**Comment by shukon** [comment]
As addition: this also sometimes happens for me when executing the `pyrfr_oob_error.py` in the examples (with 16 trees):
```
random_forest_run/pyrfr$ python pyrfr_oob_error.py 
67.8637958831
59.665806861450996
random_forest_run/pyrfr$ python pyrfr_oob_error.py 
58.9219589544
nan
random_forest_run/pyrfr$ python pyrfr_oob_error.py 
59.1355918983
61.278549997347405
```
From 1000 times refitting the forest and getting the oob, 14 times it was nan. Maybe that helps locating the issue.

**Comment by sfalkner** [comment]
should be addressed by v0.7.1
